src/gt4py/testing/suites.py

```diff
@@ -13,21 +13,33 @@
 # distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
-
+import inspect
 import sys
+import types
+from collections.abc import Mapping, Sequence
```

Comment:
`Mapping` and `Sequence` are usually symbols imported from `typing` for the specification of typing hints. For the collection abstract base classes it's better to use the full `collections.abc.Sequence` name.